### ISSUE TITLE
Fine-tuning Guidelines

### ISSUE THREAD

**Issue by lppier** [UKPLab/sentence-transformers#143]
Hi, firstly thank you so much for this library. 
Currently, we are further fine-tuning the pre-trained model against our custom STS dataset in English. 

I see that there is a validation set, and a csv showing the similarity_evaluation_results.csv is in the model folder, with epoch numbers showing the various similarity metrics.

I would like to know which epoch to stop fine-tuning. Is it recommended to look at the cosine scores in similarity_evaluation_results.csv to determine this? 

Also, is there a way where we can introduce a stopping condition currently, without modifying sentence-transformers' code?

**Comment by nreimers** [comment]
Hi @lppier 
the system only stores the model with the highest score on the evaluation set (the highest score returned by the evaluator). So finding the perfect epoch to stop is not so important, you just might waste compute time if you train for more epochs than needed.

Knowing beforehand which epoch to stop is not possible. A strategy sometimes applied is early stopping: If you do not see an improvement on the dev set for k epochs, you stop the training.

I hope  this helps.

Best
Nils Reimers

**Comment by lppier** [comment]
Thanks @nreimers for the clarity on this, very helpful.
